src/stub.c

```diff
@@ -366,24 +374,62 @@ int CALLBACK _tWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCm
    return 0;
 }
 
+static PIMAGE_NT_HEADERS retrieveNTHeader(LPVOID ptr) {
+  PIMAGE_DOS_HEADER dosHeader = (PIMAGE_DOS_HEADER)ptr;
+  return (PIMAGE_NT_HEADERS)((DWORD)dosHeader + (DWORD)dosHeader->e_lfanew);
+}
+
+/* Check whether there's an embedded digital signature */
+static char isDigitallySigned(LPVOID ptr) {
+  PIMAGE_NT_HEADERS ntHeader = retrieveNTHeader(ptr);
+  return SECURITY_ENTRY(ntHeader).Size != 0;
+}
+
+/* Find the location of ocra's signature
+   NOTE: *not* the same as the digital signature from code signing
+*/
+static LPVOID ocraSignatureLocation(LPVOID ptr, DWORD size) {
```

Comment:
Would be great to have these arguments more descriptively named, perhaps `imagePtr` and `imageSize`, or similar.

Reply:
(Elsewhere too.)

Reply:
@jakedouglas ya, i was just imitating the original code but that's no excuse heh